sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2524,6 +2528,17 @@ pi_result cuda_piQueueCreate(pi_context context, pi_device device,
     return PI_ERROR_OUT_OF_RESOURCES;
   }
 }
+pi_result cuda_piextQueueCreate(pi_context Context, pi_device Device,
+                                pi_queue_properties *Properties,
+                                pi_queue *Queue) {
+  assert(Properties);
+  // Expect flags mask to be passed first.
+  assert(Properties[0] == PI_QUEUE_FLAGS);
+  pi_queue_properties Flags = Properties[1];
+  // Extra data isn't supported yet.
+  assert(Properties[2] == 0);
```

Comment:
Correct me if I'm wrong, but this assert can be triggered by the user. Would it be better to return `PI_ERROR_INVALID_OPERATION` instead to make it a recoverable error rather than a failed assert? Note that the call should be changed to `call` rather than `call_nocheck`.

Reply:
Discussed offline. This will be done in a follow-up patch.